packages/portfolio-contract/test/portfolio.contract.test.ts

```diff
@@ -586,4 +670,206 @@ test('initial baggage', async t => {
+            dest: '@Arbitrum',
+            amount: amount_half,
+            fee: feeAcct,
+          },
+          {
+            src: '@noble',
+            dest: '@Arbitrum',
+            amount: amount_half,
+            fee: feeAcct,
+          },
+          { src: '@Arbitrum', dest: 'Aave_Arbitrum', amount, fee: feeCall },
+        ],
+      },
+    );
+
+    await eventLoopIteration();
+    await common.utils.transmitVTransferEvent('acknowledgementPacket', -1);
+    t.log('ackd send to Axelar to create account');
+
+    await simulateUpcallFromAxelar(common.mocks.transferBridge, sourceChain);
+
+    await common.utils
+      .transmitVTransferEvent('acknowledgementPacket', -1)
+      .then(() => eventLoopIteration());
+    await eventLoopIteration();
+    // await new Promise(resolve => setTimeout(resolve, 10000));
+
+    const resolverMakers = await getResolverMakers(zoe, started.creatorFacet);
+    const cctpSettlementPromise = settleCCTPWithMockReceiver(
+      zoe,
+      resolverMakers,
+      amount_half.value,
+      'eip155:42161',
+    );
+
+    await simulateCCTPAck(common.utils).finally(() =>
+      simulateAckTransferToAxelar(common.utils),
+    );
+
+    await common.utils
+      .transmitVTransferEvent('acknowledgementPacket', -1)
+      .then(() => eventLoopIteration());
+    await eventLoopIteration();
+
+    const cctpSettlementPromise2 = settleCCTPWithMockReceiver(
+      zoe,
+      resolverMakers,
+      amount_half.value,
+      'eip155:42161',
+    );
+
+    await simulateCCTPAck(common.utils).finally(() =>
+      simulateAckTransferToAxelar(common.utils),
+    );
+
+    const cctpResult = await cctpSettlementPromise;
+    const cctpResult2 = await cctpSettlementPromise2;
+    const actual = await actualP;
+
+    const result = actual.result as any;
+    t.is(passStyleOf(result.invitationMakers), 'remotable');
+
+    t.is(keys(result.publicSubscribers).length, 1);
+    const { storagePath } = result.publicSubscribers.portfolio;
+    t.log(storagePath);
+    const { contents } = getPortfolioInfo(
+      storagePath,
+      common.bootstrap.storage,
+    );
+    t.snapshot(contents, 'vstorage');
+    t.snapshot(actual.payouts, 'refund payouts');
+  },
+);
+
+test.serial(
+  'open 2 positions on an EVM chain, each from a seperate portfolio, with CCTP confirmation running in parallel',
+  async t => {
+    const { trader1, common, started, zoe, trader2 } = await setupTrader(t);
+    const { usdc, bld, poc26 } = common.brands;
+
+    // XXX: value found from reverse engineering. replace with human readable message
+    common.mocks.ibcBridge.addMockAck(
+      'eyJ0eXBlIjoxLCJkYXRhIjoiQ21vS0lTOWphWEpqYkdVdVkyTjBjQzUyTVM1TmMyZEVaWEJ2YzJsMFJtOXlRblZ5YmhKRkNneGpiM050YjNNeGRHVnpkREVTQ2pNek16TXpNekF3TURBWUF5SWdBQUFBQUFBQUFBQUFBQUFBKzdpY3dFLzdjUXNmWkZzc3Z0b001OWt5bFBRcUJYVjFjMlJqIiwibWVtbyI6IiJ9',
+      'eyJyZXN1bHQiOiJFaXNLS1M5amFYSmpiR1V1WTJOMGNDNTJNUzVOYzJkRVpYQnZjMmwwUm05eVFuVnlibEpsYzNCdmJuTmwifQ==',
+    );
```

Comment:
When is such replacement expected?
```suggestion
    // XXX: values found from reverse engineering.
    const stringToBase64 = str => Buffer.from(str).toString('base64');
    const msgDataPayload = Buffer.concat([
      Buffer.from(
        '\nj\n!/circle.cctp.v1.MsgDepositForBurn\x12E\n\fcosmos1test1\x12\n3333330000',
      ),
      Buffer.from(
        'GAMiIAAAAAAAAAAAAAAAAPu4nMBP+3ELH2RbLL7aDOfZMpT0KgU=',
        'base64',
      ),
      Buffer.from('uusdc'),
    ]);
    const msgData = {
      type: 1,
      data: msgDataPayload.toString('base64'),
      memo: '',
    };
    const ackData = {
      result: stringToBase64(
        '\x12+\n)/circle.cctp.v1.MsgDepositForBurnResponse',
      ),
    };
    common.mocks.ibcBridge.addMockAck(
      stringToBase64(JSON.stringify(msgData)),
      stringToBase64(JSON.stringify(ackData)),
    );
```

Reply:
> When is such replacement expected?

when somebody finds time, which evidently you did. That's not the style we typically use, but it's an improvement.